aten/src/ATen/native/native_functions.yaml

```diff
@@ -1802,6 +1802,14 @@
 - func: layer_norm(Tensor input, int[] normalized_shape, Tensor? weight=None, Tensor? bias=None, float eps=1e-05, bool cudnn_enable=True) -> Tensor
   use_c10_dispatcher: full
 
+- func: nanprod(Tensor self, *, ScalarType? dtype=None) -> Tensor
+  variants: function, method
+
+- func: nanprod.dim_int(Tensor self, int dim, bool keepdim=False, *, ScalarType? dtype=None) -> Tensor
+  variants: function, method
+
+- func: nanprod.int_out(Tensor self, int dim, bool keepdim=False, *, ScalarType? dtype=None, Tensor(a!) out) -> Tensor(a!)
```

Comment:
We don't support tuple of ints for `dim` argument (for `prod`). Is there any reason why it's not there? In case if we decide to add it, formulae registered in `FunctionsManual` will change.

Using only `int` for `dim` argument, will give following errors in `test_nanprod_vs_numpy`:

```
Sep 05 05:23:05 ERROR [0.003s]: test_nanprod_vs_numpy_cpu_float16 (__main__.TestTorchDeviceTypeCPU)
Sep 05 05:23:05 ----------------------------------------------------------------------
Sep 05 05:23:05 Traceback (most recent call last):
Sep 05 05:23:05   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 258, in instantiated_test
Sep 05 05:23:05     result = test(self, *args)
Sep 05 05:23:05   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 628, in only_fn
Sep 05 05:23:05     return fn(self, device, *args, **kwargs)
Sep 05 05:23:05   File "test_torch.py", line 5801, in test_nanprod_vs_numpy
Sep 05 05:23:05     self._test_nanfunc_reduction_vs_numpy(torch.nanprod, np.nanprod, device, dtype)
Sep 05 05:23:05   File "test_torch.py", line 18328, in _test_nanfunc_reduction_vs_numpy
Sep 05 05:23:05     with_keepdim=with_keepdim, with_extremal=with_extremal)
Sep 05 05:23:05   File "test_torch.py", line 18291, in _test_reduction_function_with_numpy
Sep 05 05:23:05     atol=atol, rtol=rtol, exact_dtype=exact_dtype)
Sep 05 05:23:05   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 927, in compare_with_numpy
Sep 05 05:23:05     torch_result = torch_fn(t).cpu()
Sep 05 05:23:05 TypeError: nanprod() received an invalid combination of arguments - got (Tensor, dim=tuple), but expected one of:
Sep 05 05:23:05  * (Tensor input, *, torch.dtype dtype)
Sep 05 05:23:05  * (Tensor input, int dim, bool keepdim, *, torch.dtype dtype, Tensor out)
```

This is from the definition of `_test_reduction_function_with_numpy` as we use tuple of ints to `dim` argument. Is there any workaround for this? @mruberry

Reply:
You could just write your own test or modify _test_reduction_function_with_numpy to accept single dim values, possibly by adding a kwarg to it like "dim_as_int".